delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/projects/ProjectsImpl.scala

```diff
@@ -0,0 +1,276 @@
+        fields.baseOrGenerated(ref),
+        fields.vocabOrGenerated(ref),
+        rev,
+        caller
+      )
+    ).named("updateProject", component)
+
+  override def deprecate(ref: ProjectRef, rev: Long)(implicit
+      caller: Identity.Subject
+  ): IO[ProjectRejection, ProjectResource] =
+    eval(DeprecateProject(ref, rev, caller)).named("deprecateProject", component)
+
+  override def fetch(ref: ProjectRef): UIO[Option[ProjectResource]] =
+    agg.state(ref.toString).map(_.toResource).named("fetchProject", component)
+
+  override def fetchAt(ref: ProjectRef, rev: Long): IO[ProjectRejection.RevisionNotFound, Option[ProjectResource]] =
+    eventLog
+      .fetchStateAt[ProjectState](
+        persistenceId(entityType, ref.toString),
+        rev,
+        Initial,
+        Projects.next
+      )
+      .bimap(RevisionNotFound(rev, _), _.toResource)
+      .named("fetchProjectAt", component)
+
+  override def fetch(uuid: UUID): UIO[Option[ProjectResource]] =
+    fetchFromCache(uuid).flatMap {
+      case Some(ref) => fetch(ref)
+      case None      => UIO.pure(None)
+    }
+
+  override def fetchAt(uuid: UUID, rev: Long): IO[RevisionNotFound, Option[ProjectResource]] =
+    super.fetchAt(uuid, rev).named("fetchProjectAtByUuid", component)
+
+  private def fetchFromCache(uuid: UUID): UIO[Option[ProjectRef]] =
+    index.collectFirst { case (ref, resource) if resource.value.uuid == uuid => ref }
+
+  override def list(
+      pagination: Pagination.FromPagination,
+      params: SearchParams.ProjectSearchParams
+  ): UIO[SearchResults.UnscoredSearchResults[ProjectResource]]    =
+    index.values
+      .map { resources =>
+        val results = resources.filter(params.matches).toVector.sortBy(_.createdAt)
+        SearchResults(
+          results.size.toLong,
+          results.slice(pagination.from, pagination.from + pagination.size)
+        )
+      }
+      .named("listProjects", component)
+
+  override def events(offset: Offset): fs2.Stream[Task, Envelope[ProjectEvent]] =
+    eventLog.eventsByTag(projectTag, offset)
+
+  override def currentEvents(offset: Offset): fs2.Stream[Task, Envelope[ProjectEvent]] =
+    eventLog.currentEventsByTag(projectTag, offset)
+
+  private def eval(cmd: ProjectCommand): IO[ProjectRejection, ProjectResource] =
+    for {
+      evaluationResult <- agg.evaluate(cmd.ref.toString, cmd).mapError(_.value)
+      resource         <- IO.fromOption(evaluationResult.state.toResource, UnexpectedInitialState(cmd.ref))
+      _                <- index.put(cmd.ref, resource)
+    } yield resource
+}
+
+object ProjectsImpl {
+
+  type ProjectsAggregate =
+    Aggregate[String, ProjectState, ProjectCommand, ProjectEvent, ProjectRejection]
+
+  type ProjectsCache = KeyValueStore[ProjectRef, ProjectResource]
+
+  /**
+    * The projects entity type.
+    */
+  final val entityType: String = "projects"
+
+  /**
+    * The projects tag
+    */
+  final val projectTag: String = "projects"
```

Comment:
project, according to [1.4.x](https://github.com/BlueBrain/nexus/blob/v1.4.x/delta/src/main/scala/ch/epfl/bluebrain/nexus/admin/persistence/TaggingAdapter.scala#L32)